### PR TITLE
Support running udf app on property graph with string oid type

### LINKED ISSUE #1266: [BUG] UDF App can't compile on graphs with string as OID.
**Describe the bug**
As titled.

**To Reproduce**
```python
import graphscope

from graphscope.analytical.udf import pregel
from graphscope.framework.app import AppAssets

# decorator, and assign the types for vertex data, message data.
@pregel(vd_type="double", md_type="double")
class SSSP_Pregel(AppAssets):
    @staticmethod
    def Init(v, context):
        v.set_value(1000000000.0)
    @staticmethod
    def Compute(messages, v, context):
        src_id = context.get_config(b"src")
        cur_dist = v.value()
        new_dist = 1000000000.0
        if v.id() == src_id:
            new_dist = 0
        for message in messages:
            new_dist = min(message, new_dist)
        if new_dist < cur_dist:
            v.set_value(new_dist)
            for e_label_id in range(context.edge_label_num()):
                edges = v.outgoing_edges(e_label_id)
                for e in edges:
                    v.send(e.vertex(), new_dist + e.get_int(2))
        v.vote_to_halt()
    @staticmethod
    def Combine(messages):
        ret = 1000000000.0
        for m in messages:
            ret = min(ret, m)
        return ret


graphscope.set_option(show_log = True)

sess = graphscope.session()
graph = sess.g(oid_type='string', directed=True, generate_eid=True)
my_app = SSSP_Pregel()
ret = my_app(graph)
```

### PR DESCRIPTION
**Pull request by lidongze0629**
## Related issue number

Fixes #1266

### CODE REVIEW

**Review by yecol — APPROVED** [review]

**Review by sighingnow — APPROVED** [review]

File: analytical_engine/apps/python_pie/export.h [hunk 1]
```diff
@@ -23,6 +23,27 @@ limitations under the License.
 
 namespace python_grape {
 
+#if defined(_OID_TYPE)
+using Fragment = gs::PythonPIEFragment<vineyard::ArrowFragment<
+    _OID_TYPE, vineyard::property_graph_types::VID_TYPE>>;
```
[the remark this answers is not in the document]
Comment (lidongze0629) [reply analytical_engine/apps/python_pie/export.h:28]: Define in https://github.com/alibaba/GraphScope/pull/1287/files#diff-0b42af2e75487178775dd4273cfbd96e93bad0a5fefab288e56a2f7e96c4476e

[the remark this answers is not in the document]
Comment (lidongze0629) [reply analytical_engine/apps/python_pie/export.h:28]: It looks like a little confused, we can define `_OID_TYPE` in [frame.cc](https://github.com/alibaba/GraphScope/blob/main/analytical_engine/frame/cython_app_frame.cc#L45)?

[the remark this answers is not in the document]
Comment (lidongze0629) [reply analytical_engine/apps/python_pie/export.h:27]: Remove this line and define in `frame.cc`

File: analytical_engine/frame/cython_pie_app_frame.cc [hunk 2]
```diff
@@ -39,6 +41,11 @@ using string = std::string;
 #error "Missing macro _GRAPH_TYPE or _GRAPH_HEADER"
 #endif
 
+#if defined(_OID_TYPE)
+#else
+#define _OID_TYPE vineyard::property_graph_types::OID_TYPE
+#endif
```
Comment (sighingnow) [inline analytical_engine/frame/cython_pie_app_frame.cc:47]: Just `#if !defined(_OID_TYPE)`.

**Pull request merged**